Composer/packages/client/src/pages/language-generation/code-editor.tsx

```diff
@@ -57,8 +60,18 @@ const CodeEditor: React.FC<CodeEditorProps> = props => {
 
   const currentDiagnostics = inlineMode && template ? filterTemplateDiagnostics(diagnostics, template) : diagnostics;
 
+  const onInit = (monaco: Monaco) => {
+    monacoRef.current = monaco;
+  };
+
   const editorDidMount: EditorDidMount = (_getValue, lgEditor) => {
     setLgEditor(lgEditor);
+    const m = monacoRef.current;
+    if (lgEditor && m && currentPath) {
+      lgEditor.addCommand(m.KeyCode.Escape, function() {
+        navigate(path.resolve(currentPath, '../'));
```

Comment:
See my comment about about not using path.

Reply:
Another way to approach this is not use navigate at all and instead create two actions, something like `startEditMode` and `endEditMode`. Then the business logic about what to do for those events exists in the actions layer.